monai/transforms/signal/array.py

```diff
@@ -0,0 +1,773 @@
+        frequencies: Optional[List[float]] = None,
+        fraction: Optional[List[float]] = None,
+        *args,
+        **kwargs,
+    ) -> None:
+        """
+        Args:
+            boundaries: list defining lower and upper boundaries for the square pulse magnitude,
+            lower and upper values need to be positive example : ``[0.2, 0.6]``
+            frequencies: list defining lower and upper frequencies for square pulse
+            signal generation example : ``[0.001, 0.02]``
+            fraction: list defining lower and upper boundaries for partial square pulse generation
+        """
+        super().__init__()
+        if boundaries is None or None in boundaries:
+            raise ValueError("Incompatible values: boundaries needs to be a list of float.")
+        if (boundaries is None) or (None in boundaries) or (any(x < 0 for x in boundaries)):
+            raise ValueError("Incompatible values: boundaries needs to be a list of positive float.")
+        self.boundaries = boundaries
+        self.frequencies = frequencies
+        self.fraction = fraction
+
+    def _randomize(self):
+        super().randomize(None)
+        self.magnitude = self.R.uniform(low=self.boundaries[0], high=self.boundaries[1])
+        self.fracs = self.R.uniform(low=self.fraction[0], high=self.fraction[1])
+        self.freqs = self.R.uniform(low=self.frequencies[0], high=self.frequencies[1])
+
+    def _paste_slices(self, tup):
+        pos, w, max_w = tup
+        max_w = max_w.shape[len(max_w.shape) - 1]
+        wall_min = max(pos, 0)
+        wall_max = min(pos + w, max_w)
+        block_min = -min(pos, 0)
+        block_max = max_w - max(pos + w, max_w)
+        block_max = block_max if block_max != 0 else None
+        return slice(wall_min, wall_max), slice(block_min, block_max)
+
+    def _paste(self, wall, block, loc):
+        loc_zip = zip(loc, block.shape, wall)
+        wall_slices, block_slices = zip(*map(self._paste_slices, loc_zip))
+
+        wall[:, wall_slices[0]] = block[block_slices[0]]
+
+        if wall.shape[0] == 1:
+            wall = wall.squeeze()
+        return wall
+
+    def __call__(self, signal: np.ndarray) -> np.ndarray:
+        """
+        Args:
+            signal: input 1 dimension signal to which a partial square pulse will be added
+        """
+        self._randomize()
+        if len(signal.shape) > 1:
+            length = signal.shape[len(signal.shape) - 1]
+        else:
+            signal = signal[np.newaxis, :]
+            length = signal.shape[1]
+
+        time_partial = np.arange(0, round(self.fracs * length), 1)
+        squaredpulse_partial = self.magnitude * square(self.freqs * time_partial)
+
+        loc = np.random.choice(range(length))
+        signal = self._paste(signal, squaredpulse_partial, (loc,))
+
+        if len(signal.shape) == 1:
+            signal = np.expand_dims(signal, axis=0)
+
+        return signal
+
+
+class SignalNormalize(Transform):
```

Comment:
I actually don't see what this transform does that `ScaleIntensity` doesn't already do, so I would suggest removing this one.